samples/cellular/modem_shell/src/link/link_shell.c

```diff
@@ -1076,12 +1082,17 @@ static int link_shell_edrx(const struct shell *shell, size_t argc, char **argv)
 				return -EINVAL;
 			}
 			break;
+
+		case '?':
+			mosh_error("Unknown option. See usage:");
+			goto show_usage;
+		case 'h':
 		default:
```

Comment:
With the current approach, an option which is supported by some command, but not the current command, does not cause the "Unknown option" error to be printed. One way to fix this (and probably the way `getopt_long()` is intended to be used) would be to have separate option lists for each subcommand, with only the options that are supported by that particular subcommand. However, I wonder if there would be an easier way to work around this by printing the error also from the `default` branch? Would that have any side effects?

Reply:
This has been changed pretty much as suggested throughout the MOSH